gpytorch/models/exact_prediction_strategies.py

```diff
@@ -0,0 +1,393 @@
+                self.train_train_covar.base_lazy_tensor,
+                train_interp_indices,
+                train_interp_values,
+                probe_interp_indices,
+                probe_interp_values,
+            ).evaluate()
+            test_vectors = InterpolatedLazyTensor(
+                self.train_train_covar.base_lazy_tensor,
+                train_interp_indices,
+                train_interp_values,
+                probe_test_interp_indices,
+                probe_interp_values,
+            ).evaluate()
+
+            # Get inverse root
+            train_train_covar_inv_root = train_train_covar.root_inv_decomposition(probe_vectors, test_vectors).root
+            train_train_covar_inv_root = train_train_covar_inv_root.evaluate()
+
+            # New root factor
+            root = self._exact_predictive_covar_inv_quad_form_cache(train_train_covar_inv_root, test_train_covar)
+
+            # Precomputed factor
+            if beta_features.fast_pred_samples.on():
+                inside = self.train_train_covar.base_lazy_tensor + RootLazyTensor(root).mul(-1)
+                inside_root = inside.root_decomposition().root.evaluate()
+                # Prevent backprop through this variable
+                inside_root = inside_root.detach()
+                precomputed_cache = inside_root, None
+            else:
+                # Prevent backprop through this variable
+                root = root.detach()
+                precomputed_cache = None, root
+
+        # Compute the exact predictive posterior
+        if beta_features.fast_pred_samples.on():
+            res = self._exact_predictive_covar_inv_quad_form_root(precomputed_cache[0], test_train_covar)
+            res = RootLazyTensor(res)
+        else:
+            root = left_interp(test_interp_indices, test_interp_values, precomputed_cache[1])
+            res = test_test_covar + RootLazyTensor(root).mul(-1)
+        return res, precomputed_cache
+
+
+@register_prediction_strategy(SumLazyTensor)
+class SumPredictionStrategy(DefaultPredictionStrategy):
+    def __init__(self, train_train_covar):
+        if not isinstance(train_train_covar, SumLazyTensor):
+            raise TypeError(
+                f"SumPredictionStrategy can only be used with SumLazyTensors, but got {type(train_train_covar)}"
+            )
+
+        super(SumPredictionStrategy, self).__init__(train_train_covar)
+
+    def _exact_predictive_covar_inv_quad_form_cache(self, train_train_covar_inv_root, test_train_covar):
+        test_train_covar = test_train_covar.evaluate_kernel()
+        if not isinstance(test_train_covar, SumLazyTensor):
+            return super(SumPredictionStrategy, self)._exact_predictive_covar_inv_quad_form_cache(
+                train_train_covar_inv_root, test_train_covar
+            )
+        else:
+            return tuple(
+                prediction_strategy(lazy_tensor)._exact_predictive_covar_inv_quad_form_cache(
```

Comment:
If you take the suggestion I have about caching pred_strat, then you'd probably want SumPredStrat to cache the sub-prediction strategies.